pkg/util/util.go

```diff
@@ -239,3 +244,147 @@ func SplitQualifiedName(str string) (string, string) {
+		}
+		route.Destination = ip
+		ip, err = parseIP(fields[2])
+		if err != nil {
+			return nil, fmt.Errorf("Unable to Parse IP: %s", fields[2])
+		}
+		route.Gateway = ip
+	}
+	return routes, nil
+}
+
+func parseIP(str string) (net.IP, error) {
+	bytes, err := hex.DecodeString(str)
+	if err != nil {
+		return nil, err
+	}
+	//TODO add ipv6 support
+	if len(bytes) != net.IPv4len {
+		return nil, fmt.Errorf("only IPv4 is supported")
+	}
+	bytes[0], bytes[1], bytes[2], bytes[3] = bytes[3], bytes[2], bytes[1], bytes[0]
+	return net.IP(bytes), nil
+}
+
+func isInterfaceUp(intf *net.Interface) bool {
+	if intf.Flags&net.FlagUp != 0 {
+		glog.V(4).Infof("Interface %v is up\n", intf.Name)
+		return true
+	}
+	return false
+}
+
+func getFinalIP(addrs []net.Addr) (net.IP, error) {
+	if len(addrs) > 0 {
+		for i := range addrs {
+			glog.V(4).Infof("Checking addr  %s.\n", addrs[i].String())
+			ip, _, err := net.ParseCIDR(addrs[i].String())
+			if err != nil {
+				return nil, err
+			}
+			//Only IPv4
+			//TODO : add IPv6 support
+			if ip.To4() != nil {
+				if !ip.IsLoopback() {
+					glog.V(4).Infof("IP found %v\n", ip)
+					return ip, nil
+				} else {
+					glog.V(4).Infof("Loopback found %v\n", ip)
+				}
+			} else {
+				glog.V(4).Infof("%v is not a valid IPv4 address\n", ip)
+			}
+
+		}
+	}
+	return nil, nil
+}
+
+func getIPFromInterface(intfName string) (net.IP, bool, error) {
+	intf, err := net.InterfaceByName(intfName)
+	if err != nil {
+		return nil, false, err
+	}
+	if isInterfaceUp(intf) {
+		addrs, err := intf.Addrs()
+		if err != nil {
+			return nil, false, err
+		}
+		glog.V(4).Infof("Interface %q has %d addresses :%v.\n", intfName, len(addrs), addrs)
+		finalIP, err := getFinalIP(addrs)
+		if err != nil {
+			return nil, false, err
+		}
+		if finalIP != nil {
+			glog.V(4).Infof("valid IPv4 address for interface %q found as %v.\n", intfName, finalIP)
+			return finalIP, true, nil
+		}
+	}
+
+	return nil, false, nil
+}
+
+//ChooseHostInterface is a method used fetch an IP for a daemon.
+//It uses data from /proc/net/route file.
+//For a node with no internet connection ,it returns 127.0.0.1 and the user has to set the IP to 127.0.0.1.
+//For a multi n/w interface node it returns the IP of the interface with gateway on it.
+func ChooseHostInterface() (net.IP, bool, error) {
+	zero := net.IP{0, 0, 0, 0}
+	localhost := net.IP{127, 0, 0, 1}
+	routes, err := getRoutes("/proc/net/route")
+	if err != nil {
+		return nil, false, err
+	}
+	for i := range routes {
+		//find interface with gateway
+		if routes[i].Destination.Equal(zero) {
+			glog.V(4).Infof("Default route transits interface %q\n", routes[i].Interface)
+			ip, found, err := getIPFromInterface(routes[i].Interface)
+			if err != nil {
+				return nil, false, err
+			}
+			if found {
+				glog.V(4).Infof("Choosing IP %v \n", ip)
+				return ip, true, nil
+			}
+		}
+	}
+	glog.V(4).Infof("No valid IP found\n")
+	return localhost, false, nil
```

Comment:
I think for now we should return an error, which means no need for localhost variable